src/thunderbots/software/radio_communication/mrf/usb/transfer.cpp

```diff
@@ -97,12 +97,13 @@ void USB::Transfer::result() const
 void USB::Transfer::submit()
 {
     assert(!submitted_);
-    check_fn("libusb_submit_transfer", libusb_submit_transfer(transfer),
-             transfer->endpoint);
     submitted_         = true;
     done_              = false;
     stall_retries_left = retry_on_stall_ ? 30 : 0;
     ++device.submitted_transfer_count;
+
+    check_fn("libusb_submit_transfer", libusb_submit_transfer(transfer),
```

Comment:
This might have been the actual reason for the race condition, the transfer completing in the other thread before `done_` gets set -.-